From d728eec1bcf0f360aea989a09eeb6fc47b4704d1 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Thu, 21 May 2020 16:46:37 +0200
Subject: [PATCH] Increase default u-boot environment size

After balenaOS 2.39.0 u-boot will not be able
to load the default environment as it exceeds the
default size:
*** Error - default environment is too large

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 include/configs/tegra-common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/tegra-common.h b/include/configs/tegra-common.h
index f2cdd9c019..a27cfa28ed 100644
--- a/include/configs/tegra-common.h
+++ b/include/configs/tegra-common.h
@@ -25,7 +25,7 @@
 #define CONFIG_CMDLINE_TAG		/* enable passing of ATAGs */
 
 /* Environment */
-
+#define CONFIG_ENV_SIZE                        0x3000  /* Total Size Environment */
 /*
  * NS16550 Configuration
  */
-- 
2.17.1

